refactor: implement the new bzlmod api

thesayyn opened the pull request:
Fixes #123 #56 #66 https://github.com/bazel-contrib/rules_distroless/issues/124

thesayyn committed:
refactor: implement the new bzlmod api

ryanli commented on apt/extensions.bzl:
```diff
@@ -83,33 +160,33 @@ Here's an example how to create a Debian repo:
-
-`apt.install` will parse the manifest and will fetch and install the packages
-for the given architectures in the Bazel repo `@<NAME>`.
+`apt.install` will install generate a package repository for each package and architecture
+combination in the form of `@<TARGET_RELEASE>_<PKG_NAME>_<PKG_ARCH>`.
```
Could there be a way to specify the package version too? Regenerating the lock file currently always use the latest version but sometimes we could want older ones for compatibility.

thesayyn committed:
more fixes

baxelrod-bdai commented:
I'd find this feature very useful.  I hope it gets merged soon!

thesayyn commented:
Still working on this, there will be some breaking changes but this will eventually land./

thesayyn committed:
latest

thesayyn committed:
lts

thesayyn committed:
lts

thesayyn committed:
fix symlinks

thesayyn committed:
rm unused

thesayyn committed:
fix

thesayyn committed:
lts

thesayyn committed:
example

thesayyn committed:
lts

thesayyn committed:
lts

thesayyn committed:
lts

thesayyn committed:
so_library

thesayyn committed:
make it green

blorente committed:
fix: Allow multiple indices in foreign_symlinks (#205)

blorente committed:
cleanup: Remove log lines (#206)

thesayyn committed:
suites

blorente committed:
fix: Allow multiple cc_imports per pkgconfig file (#208)

Some packages, like `krb5-multidev`, include several .so files to link
against in their pkg-config files:
```
prefix=/usr
exec_prefix=${prefix}
libdir=${prefix}/lib/x86_64-linux-gnu/mit-krb5
includedir=${prefix}/include/mit-krb5

defccname=FILE:/tmp/krb5cc_%{uid}
defktname=FILE:/etc/krb5.keytab
defcktname=FILE:/etc/krb5/user/%{euid}/client.keytab

Name: mit-krb5
Description: An implementation of Kerberos network authentication
Version: 1.20.1
Cflags: -isystem ${includedir}
Libs: -L${libdir} -lkrb5 -lk5crypto -lcom_err
Libs.private: -lkrb5support
```

So, instead of assuming that we have one cc_library per pkgconfig file,
we just collect all the declarations and create one `cc_import` target
for each.

Some design considerations:

### Does this violate the private-ness of `Libs.private`?

All `cc_imports` we create have private visibility already, so I don't
think it does.

### Why not depend on other debian packages exporting those libraries?

Since those files are already in the package
(https://packages.debian.org/sid/amd64/krb5-multidev/filelist), I assume
the intention is to link against the `.so` files distributed in the
`krb-multidev` package, and not to pull other deb packages. This could
be false, of course, if a user has already installed those packages and
for some reason the linker finds their SO files first, but I think
that's a bug in the definition of the krb5-multidev package, not this
implementation.

Plus, it has the added benefit that, if the debian package in question
doesn't include an appropriate .so file, we just don't create the
cc_import.

blorente committed:
fix: Several fixes for dependency resolution (#211)

We should probably rebase and merge this one, instead of squashing.

thesayyn committed:
cleanup

thesayyn committed:
fix external_include_paths feature
